src/main/java/VASSAL/build/module/map/KeyBufferer.java

```diff
@@ -100,8 +109,12 @@ public void mousePressed(MouseEvent e) {
     // Don't clear the buffer until we find the clicked-on piece
     // Because selecting a piece affects its visibility
     EventFilter filter = null;
+    BandSelectType bandSelect = BandSelectType.NONE;
     if (p != null) {
-      filter = (EventFilter) p.getProperty(Properties.SELECT_EVENT_FILTER);
+      filter = (EventFilter) p.getProperty(Properties.SELECT_EVENT_FILTER);      
+      if (!e.isPopupTrigger() && Boolean.TRUE.equals(p.getProperty(Properties.NON_MOVABLE))) {
```

Comment:
Um, does this comment belong in this PR? I think this is just from when I re-based the PR on master. 

The answer to your question is: because if it's a right-click then it's not a band-select and therefore we proceed with normal clicking-on-things-and-putting-up-the-popup-menu.

Reply:
I'm not sure why these lines show up as a change on _this_ PR. That said, I think this is wrong---`isPopupTrigger()` checks for something _waves hands_ which is not quite the same as a right click, and it got my attention because nothing in this file raises a context menu. I think we might want to check that the mouse button isn't the left here instead.

Reply:
Oh right isPopupTrigger() means is THIS event going to open the context menu, and runs into the whole issue of some platforms do the trigger on "Pressed" and others do it on "Released" and when you're in the wrong method for your platform it doesn't tell you a damn thing. So this might produce hinky results on Linux which I *think* is the opposite whateverness from Windows. I will make a new PR and improve it.

Meanwhile no idea why THIS PR. The commands I issued to rebase this PR on master were:
git rebase --onto master HTMLChatter
git pull
git push -u origin HTMLChatter

Reply:
I imagine that means when it comes time to merge this you'll want to do one of those Squash-N-Crunch kinds of merge.

Reply:
IIRC, the weird one for popup triggers is the Mac?

Reply:
Upon reading up on it (because why not, "mouse is my thing"), it turns out that Windows is "one way" and Mac & Linux (both descendants of Unix) are "the other way".